empyrical/utils.py

```diff
@@ -447,3 +463,72 @@ def default_returns_func(symbol, start=None, end=None):
         rets = get_symbol_returns_from_yahoo(symbol, start=start, end=end)
 
     return rets[symbol]
+
+
+def rolling_window(array, length):
+    """
+    Restride an array of shape
+
+        (X_0, ... X_N)
+
+    into an array of shape
+
+        (length, X_0 - length + 1, ... X_N)
+
+    where each slice at index i along the first axis is equivalent to
+
+        result[i] = array[length * i:length * (i + 1)]
+
+    Parameters
+    ----------
+    array : np.ndarray
+        The base array.
+    length : int
+        Length of the synthetic first axis to generate.
+
+    Returns
+    -------
+    out : np.ndarray
+
+    Example
+    -------
+    >>> from numpy import arange
+    >>> a = arange(25).reshape(5, 5)
+    >>> a
+    array([[ 0,  1,  2,  3,  4],
+           [ 5,  6,  7,  8,  9],
+           [10, 11, 12, 13, 14],
+           [15, 16, 17, 18, 19],
+           [20, 21, 22, 23, 24]])
+
+    >>> rolling_window(a, 2)
+    array([[[ 0,  1,  2,  3,  4],
+            [ 5,  6,  7,  8,  9]],
+    <BLANKLINE>
+           [[ 5,  6,  7,  8,  9],
+            [10, 11, 12, 13, 14]],
+    <BLANKLINE>
+           [[10, 11, 12, 13, 14],
+            [15, 16, 17, 18, 19]],
+    <BLANKLINE>
+           [[15, 16, 17, 18, 19],
+            [20, 21, 22, 23, 24]]])
+    """
+    orig_shape = array.shape
+    if not orig_shape:
+        raise IndexError("Can't restride a scalar.")
+    elif orig_shape[0] < length:
```

Comment:
Might want to also check the `length == 0` case too

Reply:
Also when length is a float, there's strange behavior. Using `a` from the docstring:

```
>>> rolling_window(a, 2.01)
array([[[ 0,  1,  2,  3,  4],
        [ 5,  6,  7,  8,  9]],

       [[ 5,  6,  7,  8,  9],
        [10, 11, 12, 13, 14]],

       [[10, 11, 12, 13, 14],
        [15, 16, 17, 18, 19]]])
```

Reply:
what needs to be done in the `length == 0` case? You will get an array with a dimension of size 0, but that is fine.

Also, I think the second thing is fine, we say that it takes an `int`, so I wouldn't expect it to work with non-int fields.